permcheck.go

```diff
@@ -17,8 +17,20 @@ type PermChecker struct {
 	cache *GroupCache
 }
 
-// NewPermChecker returns a permission checker
-// that uses the given identity client to check permissions.
+// NewPermChecker returns a permission checker that uses the given
+// identity client to check permissions. The following rules apply when
+// checking if a user is a member of a name in an ACL. The PermChecker
+// chooses whether to authorize the user u with respect to the ACL name n
+// by following these rules:
+//
+// - If u is identical to n, authorization is granted.
```

Comment:
What about:
user who is only allowed if the acl is "who" or "everyone"
user who@azure is only allowed if the acl is "who@azure" or "who@everywhere" or "everyone@azure", "everyone@everywhere"
Isn't it easier to explain? Also, it means that you have to specify something more to open up privileges, "everyone", "everywhere", and by default they are more constrained rather that the other way around (having to remember to add no-domain).